### ISSUE TITLE
Need an option to pass generic props to components

### ISSUE THREAD

**Issue by yanivefraim** [wix/wix-style-react#10]
This is important, for example, if I want to use a selector for my component, I can do something like:

```
<WixStyle.Button data-hook="my-button" />
```

And then in my test I can do something like:

```
myComponent.find('[data-hook="my-button"]')
```

I will add a PR soon

**Comment by yanivefraim** [comment]
The solution is currently in a form of a testkit instead.

**Comment by 3LOK** [comment]
To summarize what we discussed: The way I see it, exposing anything that has to do with the DOM implementation of my React component, is much like exposing private functions used in a class. If I give any way to manipulate them (but giving a way to find them in a selector), I'm basically binding myself to a very specific DOM structure implementation of my component, which is very limiting in terms of future refactors.

As such, we decided the right approach is to expose a test-kit driver utility for each of the different components. These driver utilities will allow simulating different events, given a unique component id.

See for example [this PR](https://github.com/wix/wix-style-react/pull/13).
